packages/admin/cms-admin/src/generator/future/utils/generateFieldList.ts

```diff
@@ -0,0 +1,69 @@
+import { IntrospectionField, IntrospectionQuery, IntrospectionType } from "graphql";
+import objectPath from "object-path";
+
+type FieldsObjectType = { [key: string]: FieldsObjectType | boolean | string };
+const recursiveStringify = (obj: FieldsObjectType, objectSuffix?: string): string => {
+    let ret = "";
+    let prefixField = "";
+    for (const key in obj) {
+        const valueForKey = obj[key];
+        if (typeof valueForKey === "boolean") {
+            ret += `${prefixField}${key}`;
+        } else if (typeof valueForKey === "string") {
+            ret += `${prefixField}${key}${valueForKey}`;
+        } else {
+            ret += `${prefixField}${key}${objectSuffix ? objectSuffix : ""} { ${recursiveStringify(valueForKey)} }`;
+        }
+        prefixField = " ";
+    }
+    return ret;
+};
+
+export function getRootProps(fields: string[]): string[] {
```

Comment:
Why is this called root props?

Reply:
because the code-snippets we create only care about the props on the first level, and my first naming idea was root props (props of the root-object). we could also call it `getFirstLevelProps`?